pandas/core/reshape/merge.py

```diff
@@ -952,6 +959,51 @@ def _validate_specification(self):
         if len(self.right_on) != len(self.left_on):
             raise ValueError("len(right_on) must equal len(left_on)")
 
+    def _validate(self):
+        # Get merging series:
```

Comment:
simply use ``self.left_join_keys`` and ``self.right_join_keys`` which are already lists of arrays (1 or more). No need to repeat this logic.